salt/returners/splunk.py

```diff
@@ -48,6 +48,14 @@ def returner(ret):
     return _send_splunk(ret)
 
 
+def event_return(events):
+    """
+    Return events to Splunk via the HTTP Event Collector
+    """
+    for event in events:
+        _send_splunk(event)
```

Comment:
Can you refactor `_send_splunk` to reduce the overhead of sending payloads? You can use the `batchEvent` if you want, but be sure to test it as I see a one or two bugs in it.

Here's what I suggest:

Take the 3 actions in `_send_splunk`: 

 - create an `http_event_collector`
 - preparing a payload
 - sending the payload through the event collector

Refactor the first 2 actions:

 - create a function that creates an event collector using opts (maybe `_event_collector(opts)`)
 - create a function that prepares a payload (maybe `_prepare_payload(event, opts)`)

Then:

1. Modify `returner` to be something like:

```python
def returner(ret):
    opts = _get_options()
    event_collector = _event_collector(opts)
    payload = _prepare_payload(ret, opts)
    event_collector.sendEvent(payload)
```

2. Modify `event_return` to be something like:

```python
def event_return(events):
    opts = _get_options()
    event_collector = _event_collector(opts)
    for event in events:
        payload = _prepare_payload(event, opts)
        event_collector.sendEvent(payload)
```

3. Delete `_send_splunk`.

Reply:
Thank you for the detailed review. I've refactored as directed and removed the  `batchEvent` related code in `http_event_collector` as it is unused.

Reply:
Thanks, it looks great!